src/shipper/elastic.ts

```diff
@@ -65,54 +54,59 @@ export class ElasticSearch {
    */
   queue(logObj: LogObject, prefix: string, indexType: LogShipperConfigIndexDate): void {
     const indexName = this.getIndexName(logObj, prefix, indexType);
-    this.indexes[indexName] = true;
-    this.body.push({ index: { _index: indexName, _type: '_doc', _id: logObj['@id'] } });
-    this.body.push(logObj);
+    this.indexes.set(logObj['@id'], indexName);
```

Comment:
I couldn't think of a better way of doing this, its kinda expensive to lookup this information on a log by log basis, so seems safest to cache it somewhere, does mean we need to do a `!=null` check when fetching it back out though (should never be null)

and these id's should be globally unique, they are the uniqueId to store inside of elasticsearch